content/sensu-go/5.17/reference/secret-providers.md

```diff
@@ -0,0 +1,297 @@
+    }
+  },
+  "max_retries": 2,
+  "timeout": "20s",
+  "rate_limiter": {
+    "limit": 10.0,
+    "burst": 100
+  }
+}
+{{< /highlight >}}
+
+### Metadata attributes
+
+name         |      |
+-------------|------
+description  | Provider name used internally by Sensu.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"name": "vault"{{< /highlight >}}
+
+### Spec attributes
+
+client       | 
+-------------|------ 
+description  | Map that includes secrets provider configuration [client attributes][12].
+required     | true
+type         | Map of key-value pairs
+example      | {{< highlight shell >}}
+"client": {
+  "address": "https://vaultserver.example.com:8200",
+  "token": "VAULT_TOKEN",
+  "version": "v1",
+  "tls": {
+    "ca_cert": "/etc/ssl/certs/vault_ca_cert.pem"
+}
+{{< /highlight >}}
+
+max_retries  | 
+-------------|------ 
+description  | Number of times to retry connecting to the vault provider.
+required     | true
+type         | Integer
+default      | 2
+example      | {{< highlight shell >}}"max_retries": 2{{< /highlight >}}
+
+timeout      | 
+-------------|------ 
+description  | Provider connection timeout (hard stop).
+required     | false
+type         | String
+default      | 60s
+example      | {{< highlight shell >}}"timeout": "20s"{{< /highlight >}}
+
+rate_limiter | 
+-------------|------ 
+description  | Maximum rate and burst limits for the secrets API.
+required     | false
+type         | Map of key-value pairs
+example      | {{< highlight shell >}}
+"rate_limiter": {
+  "limit": 10.0,
+  "burst": 100
+}
+{{< /highlight >}}
+
+#### Client attributes
+
+address      | 
+-------------|------ 
+description  | Vault server address.
+required     | true
+type         | String
+example      | {{< highlight shell >}}
+"address": "https://vaultserver.example.com:8200"
+{{< /highlight >}}
+
+token        | 
+-------------|------ 
+description  | Vault token to use for authentication.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"token": "VAULT_TOKEN"{{< /highlight >}}
+
+version      | 
+-------------|------ 
+description  | HashiCorp Vault [HTTP API version][14].
+required     | true
+type         | String
+example      | {{< highlight shell >}}"version": "v1"{{< /highlight >}}
+
+tls          | 
+-------------|------ 
+description  | TLS object. Vault only works with TLS configured. You may need to set up a CA cert if it is not already stored in your operating system's trust store. To do this, set the TLS object, and provide the `ca_cert` path. You may also need to set up a client cert, client key, or cname.
+required     | true
+type         | Map of key-value pairs
+example      | {{< highlight shell >}}
+"tls": {
+  "ca_cert": "/etc/ssl/certs/vault_ca_cert.pem",
+  "client_cert": "/etc/ssl/certs/vault_cert.pem",
+  "client_key": "/etc/ssl/certs/vault_key.pem",
+  "cname": "vault_client.example.com"
```

Comment:
Unlike the other parameters in this TLS object, it's not immediately clear to me what the `cname` parameter does.

Reply:
Yes -- we do not seem to mention it elsewhere in our docs. @echlebek is it OK to remove the cname item?

Reply:
Update: I searched the Vault docs this morning and didn't find anything about using 'cname'
I did find this: https://github.com/hashicorp/vault/issues/3709

Reply:
Update 2: then again, I don't see it in Cyril's PR: https://github.com/sensu/sensu-staging/pull/188/files/6c443c8f4afa36d898856453cf91d2c20d4dd47e..26f4ea6f851c4a66693d72af142619bdd8b7d610

So now I'm back to wondering if we can remove the cname item.

Reply:
The cname corresponds to the name parameter mentioned here: https://www.vaultproject.io/api/auth/cert/index.html#parameters-7

Reply:
Got it, thank you. Do we need to explain why it is `cname` in our docs and `name` in the Vault docs?